### ISSUE TITLE
SPIKE - phone number filter

### ISSUE THREAD

**Issue by blaughman** [usdoj-crt/crt-portal-management#1341]
## User story:
As a CRT user, I want a filter to locate a record using the correspondent's phone number so that I can find reports more efficiently.

## Objective:
More useful tools for our users.

## Key result:
Increase in staff efficiency and turnaround time.

## Ticket background:
DRS and VOT take in a lot of phone reports.  At times, they experience difficulty in hearing/spelling a person's name, so they need to locate their record with a phone number.

## Technical questions / concerns
- Is there an API we could use?
- Could we have the filter omit any non-numerical characters?  For instance, if someone put in a phone number on the form of (202) 555-5555 could the filters omit the '(' ')' '<space>' and '-'

**Comment by blaughman** [comment]
Current error message on Production

<img width="214" alt="image" src="https://user-images.githubusercontent.com/52507395/177623909-8d119817-2109-4542-a7dd-821809c633e6.png">

**Comment by billyhunt** [comment]
I think we could definitely improve the phone number input to make it more usable, and in line with the us web design system.  

Another thing we could do is add a new filter to the search page for phone number.  That filter would, ahem, filter out any non numerical characters (eg "(", "-") so it would search for the correct number.  

I propose we make two tickets.  One to update the fields on the Pro Form and Report Page, and another to add the field to the search page.  

Make sense?

**Comment by chinelo-ikejimba** [comment]
> I think we could definitely improve the phone number input to make it more usable, and in line with the us web design system.
> 
> Another thing we could do is add a new filter to the search page for phone number. That filter would, ahem, filter out any non numerical characters (eg "(", "-") so it would search for the correct number.
> 
> I propose we make two tickets. One to update the fields on the Pro Form and Report Page, and another to add the field to the search page.
> 
> Make sense?

Yep that makes sense @billyhunt. For great UX, let's include a [country dropdown](https://intl-tel-input.com/) that way staff can verbally check with caller's that the country matches the telephone number the caller provided.

**Example image**
<img width="428" alt="Screen Shot 2022-07-06 at 4 32 38 PM" src="https://user-images.githubusercontent.com/92807659/177637933-6fc53648-6d1c-47a6-badc-0476ced00acf.png">

**Comment by billyhunt** [comment]
You read my mind!

**Comment by randyabramson** [comment]
7/15/22 - Per Billy, the simple search for the raw numbers is working, but we don't have it working for parenthesis, hyphens, spaces, etc.. He's going to keep working on this.

**Comment by billyhunt** [comment]
Just to add a little clarification, here are the notes and limitations on running phone number searches. from PR [1214](https://github.com/usdoj-crt/crt-portal/pull/1214)

Notes:

* This search works by breaking a number into numerical chunks, and filtering on those chunks.  "123-456-7890" would filter anything that had "123" AND "456" AND "7890".  So "123-456-7890", "123 456 7890", "(123) 456-7890", and "1.123.456.7890x4567" would all be returned from this search.
* I added phone numbers to the factory, so add some new reports with ```docker-compose run web python crt_portal/manage.py create_mock_reports 1000``` for easier testing.
* This design has been approved by Chinelo. We might need to alert intake specialists that they need to break up numbers, but since this is the natural behavior when writing phone numbers, she felt it was not worth cluttering the interface with help text.

Limitations:
* This search removes all non digit characters.  So if an intake specialist enters the search "123 (*&(*&^(&HJDAJIHDS)" in the phone number field, it is only going to search for "123'
* you need to separate your numbers into blocks with spaces or any other character for it to search the phone number.  "1234567890" would NOT find "123-456-7890".  The reason for this is the search would be computationally expensive and return unwanted results if I break up the search without knowing the groupings.  
* Because I am searching the number against blocks of digits, if the blocks are identical, its the same as searching just the single block.  EG "123 123" is the same search as "123"

**Comment by billyhunt** [comment]
TLDR: Spaces, "(" "-", really anything can break up the numbers and it will work fine.

**Comment by billyhunt** [comment]
@blaughman does this ticket have anything left on it?

**Comment by blaughman** [comment]
@billyhunt Closing this card.  The card for the front end is already created.
